app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :post_owner?, only: %i[edit update destroy]
+
+  def index
+    @post = Post.new
+    @posts = Post.all
+  end
+
+  def show; end
+
+  def new
+    @post = Post.new
+  end
+
+  def edit; end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      flash[:notice] = 'Post was successfully created.'
+      redirect_to @post
+    else
+      render :new
+    end
+  end
+
+  def update
+    render :edit unless @post.update(post_params)
+  end
```

Comment:
- [ ] **[OPTIONAL]** After updating a post it looks like the page freezes and the user doesn't know whether the post updates or not. Please let's add some flash and redirect somewhere else if the post is updated.